src/hpx_init.cpp

```diff
@@ -584,6 +595,19 @@ namespace hpx
 
             int result = 0;
             try {
+                #if (HPX_HAVE_DYNAMIC_HPX_MAIN != 0) && \
+                    (defined(__linux) || defined(__linux__) || defined(linux))
+                // make sure the runtime system is not initialized
+                // after its activation from int main()
+                if(get_runtime_ptr() != nullptr &&
```

Comment:
Do we have to check for `get_runtime_ptr() != nullptr` twice?

Reply:
Yes. `include_libhpx_wrap` will only define whether the route of initialization is through main or not. To know whether hpx has already initialized, we will need the help of `get_runtime_ptr()`. I'll merge the 2 if statements to make it more clear.